data/model/repository.py

```diff
@@ -672,3 +675,48 @@ def mark_repository_for_deletion(namespace_name, repository_name, repository_gc_
         callback(namespace_name, repository_name)
 
     return marker.id
+
+
+def get_repository_size_and_cache(repo_id: int):
+    query = (
+        RepositorySize.select(Repository.name, Repository.id, RepositorySize.size_bytes)
+        .join(Repository)
+        .where(Repository.id == repo_id)
+    ).get()
+
+    if query is None:
+        force_cache_repo_size(repo_id)
+
+    # TODO: this will likely fail if we have to force cache
+    return {
+        "repository_name": query.repository.name,
+        "repository_id": query.repository.id,
+        "repository_size": query.size_bytes,
+    }
```

Comment:
This doesn't match your type hint signature (int). Either return just the int, or the `query` itself if you need access to the parent repo.